tests/integration/devfile/cmd_devfile_exec_test.go

```diff
@@ -25,25 +26,58 @@ var _ = Describe("odo devfile exec command tests", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("When devfile exec command is executed", func() {
+	When("a component is created", func() {
 
-		It("should execute the given command successfully in the container", func() {
-			utils.ExecCommand(commonVar.Context, cmpName)
-			podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
-			listDir := commonVar.CliRunner.ExecListDir(podName, commonVar.Project, "/projects")
-			Expect(listDir).To(ContainSubstring("blah.js"))
-		})
+		BeforeEach(func() {
+			args := []string{"create", "nodejs", cmpName, "--context", commonVar.Context}
+			helper.Cmd("odo", args...).ShouldPass()
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
 
-		It("should error out when no command is given by the user", func() {
-			utils.ExecWithoutCommand(commonVar.Context, cmpName)
 		})
 
-		It("should error out when a invalid command is given by the user", func() {
-			utils.ExecWithInvalidCommand(commonVar.Context, cmpName, "kube")
+		It("should error out when a component is not present or when a devfile flag is used", func() {
+			args := []string{"exec", "--context", commonVar.Context}
```

Comment:
I think something like the following would make sense in understanding the test easily. What do you think?
```go
		It("should error out when a component is not present or when a devfile flag is used", func() {
			By("the component is not present")
			args := []string{"exec", "--context", commonVar.Context}
			args = append(args, []string{"--", "touch", "/projects/blah.js"}...)
			helper.Cmd("odo", args...).ShouldFail()

			By("the devfile flag is used")
			args = []string{"exec", "--context", commonVar.Context, "--devfile", "invalid.yaml"}
			args = append(args, []string{"--", "touch", "/projects/blah.js"}...)
			helper.Cmd("odo", args...).ShouldFail()
		})
```